### PR TITLE
fix #19250 ORC AssertionDefect not containsManagedMemory(n.typ)

### LINKED ISSUE #19250: ORC AssertionDefect `not containsManagedMemory(n.typ)`
The following code failed to compile with `--gc:orc` 

### Example
```nim
type
  Bar[T] = object
    err: proc(): string

  Foo[T] = object
    run: proc(): Bar[T]

proc bar[T](err: proc(): string): Bar[T] =
  assert not err.isNil
  Bar[T](err: err)

proc foo*(): Foo[char] = 
  result.run = proc(): Bar[char] =
    
    # works
    # result = Bar[char](err: proc(): string = "x")
    
    # not work
    result = bar[char](proc(): string = "x")

proc bug*[T](fs: Foo[T]): Foo[T] =
  result.run = proc(): Bar[T] =
    let res = fs.run()
    
    # works
    # var errors = @[res.err] 
    
    # not work
    var errors: seq[proc(): string]
    errors.add res.err
    
    return bar[T] do () -> string:
      for err in errors:
        result.add res.err()

assert bug(foo()).run().err() == "x"
```

### Current Output

Fail to compile with the following assertion defect. 

```
Error: unhandled exception: injectdestructors.nim(467, 13) `not containsManagedMemory(n.typ)`  [AssertionDefect]
```

### Expected Output

should compile.

### Possible Solution

* There are two places commented in the above example showing a little change could work-around the assertion. But I cannot make sense with those workarounds.  Probably some assumptions inside orc do not meet.

### Additional Information

run with

`nim r --gc:orc test.nim`

Tested with 1.6.0 and devel of nim.

```
$ nim -v
Nim Compiler Version 1.7.1 [Linux: amd64]
Compiled at 2021-12-14
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 0ff4b2ba7ee74ed2758b0eb5992e4ccab7433952
active boot switches: -d:release
```
```
$ nim -v
Nim Compiler Version 1.6.0 [Linux: amd64]
Compiled at 2021-10-19
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 727c6378d2464090564dbcd9bc8b9ac648467e38
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by bung87**
fix #19250

### CODE REVIEW

File: tests/arc/topt_no_cursor.nim [hunk 1]
```diff
@@ -101,6 +105,7 @@ try:
           `=destroy`(splitted)
 finally:
   `=destroy_1`(lan_ip)
+-- end of expandArc ------------------------
```
Comment (Araq) [inline tests/arc/topt_no_cursor.nim]: Why would this line be new? Something is messed up here.

Comment (bung87) [reply tests/arc/topt_no_cursor.nim]: but here is expected, it should be present at each injectDestructorCalls

Comment (Araq) [reply tests/arc/topt_no_cursor.nim]: Is this a testament bug then? How come it ever was green before?

Comment (bung87) [reply tests/arc/topt_no_cursor.nim]: it's messed up before, https://github.com/nim-lang/Nim/commit/f32ffb6ed821cc01e52c48181a4caa15e73c0362 line 109 followed `-- end of expandArc ------------------------` before, and then it doesn't.

Comment (Araq) [reply tests/arc/topt_no_cursor.nim]: That's a severe testament issue. Can you investigate?

Comment (bung87) [reply tests/arc/topt_no_cursor.nim]: sure, I'll try

Comment (bung87) [reply tests/arc/topt_no_cursor.nim]: it's just mis used test, https://github.com/nim-lang/Nim/pull/20791

File: tests/arc/topt_no_cursor.nim [hunk 2]
```diff
@@ -135,16 +140,19 @@ if dirExists(this.value):
...
-    :tmpD_1)
-  `=destroy`(:tmpD_3)
+    wasMoved(:tmpD_3)
+    `=copy`(:tmpD_3,
+      :tmpD_2 = splitDrive do:
+        wasMoved(:tmpD_1)
+        `=copy`(:tmpD_1, this.value)
+        wasMoved(:tmpD_4)
+        `=copy`(:tmpD_4, :tmpD_1)
+        :tmpD_4
```
Comment (Araq) [inline tests/arc/topt_no_cursor.nim:144]: How do we know this does not introduce a new leak? tmpD_4 is a copy but not destroyed!

**Pull request closed without merging**